src/Features/LanguageServer/Protocol/Features/Options/CompletionOptionsStorage.cs

```diff
@@ -21,7 +20,8 @@ public static CompletionOptions GetCompletionOptions(this IGlobalOptionService o
                 // If the option is null (i.e. default) or 'true', then we want to trigger completion.
                 // Only if the option is false do we not want to trigger.
                 LanguageNames.VisualBasic => options.GetOption(TriggerOnDeletion, language) is not false,
-                _ => throw ExceptionUtilities.Unreachable()
+                // Other languages might want to get completion options, like Razor, just forward the call to option service when it happens.
+                _ => options.GetOption(TriggerOnDeletion, language),
```

Comment:
Should this just be the same as the VB branch, and if so can we just delete the VB branch?

Reply:
`null` means razor could decided the behavior by itselfs
For VB:
true -> true,
null -> true,
false -> false.

I  don't know if razor wants the same behavior, but I intend to keep Vb seperate here